service/history/workflow/update/update_test.go

```diff
@@ -543,3 +543,43 @@ func TestRollbackCompletion(t *testing.T) {
 		require.NoError(t, err, "update should be back in Accepted state")
 	})
 }
+
+func TestRejectionWithAcceptanceWaiter(t *testing.T) {
+	t.Parallel()
+	var (
+		ctx      = context.Background()
+		store    = mockEventStore{Controller: effect.Immediate(ctx)}
+		updateID = t.Name() + "-update-id"
+		upd      = update.New(updateID, ignoreCompletion)
+		req      = updatepb.Request{
+			Meta:  &updatepb.Meta{UpdateId: updateID},
+			Input: &updatepb.Input{Name: "not_empty"},
+		}
+		rej = updatepb.Rejection{
+			RejectedRequestMessageId: "not_empty",
+			RejectedRequest:          &req,
+			Failure: &failurepb.Failure{
+				Message: "intentional falure from " + t.Name(),
+			},
+		}
+	)
+	ch := make(chan any, 1)
+	go func() {
+		fail, err := upd.WaitAccepted(ctx)
+		if err != nil {
+			ch <- err
+			return
+		}
+		ch <- fail
+	}()
+	t.Log("give 5ms for the goro to get to the Future.Get call in WaitAccepted")
+	time.Sleep(5 * time.Millisecond)
+
+	t.Log("deliver request and rejection messages")
+	require.NoError(t, upd.OnMessage(ctx, &req, store))
+	require.NoError(t, upd.OnMessage(ctx, &rej, store))
+
+	outcome, ok := (<-ch).(*updatepb.Outcome)
+	require.True(t, ok)
```

Comment:
Can you add the outcome as a message here? Otherwise the error will get dropped.
```suggestion
        result := <-ch
	outcome, ok := result.(*updatepb.Outcome)
	require.True(t, ok, result)
```

Reply:
There might also be something like require.As, but I'm not sure

Reply:
It's require.TypeIs but it's not all that useful since you still have to do the cast anyway. I added the type returned from the channel to the error message if require.True fails.